src/dmd/backend/gother.c

```diff
@@ -1324,11 +1324,18 @@ void rmdeadass()
                             continue;
                         //printf("assnod[%d]: ",j); WReqn(n); printf("\n");
                         //printf("\tPOSS\n");
-                        /* If not positively dead but v is live on a    */
-                        /* successor to b, then v is live.              */
-                        //printf("\tDEAD=%d, live=%d\n",vec_testbit(j,DEAD),vec_testbit(v->Ssymnum,b->Boutlv));
-                        if (!vec_testbit(j,DEAD) && vec_testbit(v->Ssymnum,b->Boutlv))
-                                continue;
+                        if (!vec_testbit(j,DEAD))
+                        {
+                            /* If not positively dead but v is live on a    */
+                            /* successor to b, then v is live.              */
+                            //printf("\tDEAD=%d, live=%d\n",vec_testbit(j,DEAD),vec_testbit(v->Ssymnum,b->Boutlv));
+                            if (vec_testbit(v->Ssymnum,b->Boutlv))
+                                    continue;
+                            /* Do not remove assignments to ambiguous values as
+                             * they might be reached trough a pointer */
+                            if (!(v->Sflags & SFLunambig))
```

Comment:
I suspect the bug is elsewhere, as the testbit on b->Boutlv should take into account ambiguous values

Reply:
The variable doesn't escape the BB, that's why it's missing from there.
I've pushed another fix that's simpler and hopefully more correct.